src/query/phrase_query/phrase_weight.rs

```diff
@@ -88,10 +115,26 @@ impl PhraseWeight {
 
 impl Weight for PhraseWeight {
     fn scorer(&self, reader: &SegmentReader, boost: Score) -> crate::Result<Box<dyn Scorer>> {
-        if let Some(scorer) = self.phrase_scorer(reader, boost)? {
-            Ok(Box::new(scorer))
-        } else {
-            Ok(Box::new(EmptyScorer))
+        match self.get_scorer_params(reader, boost)? {
+            Some((term_postings_list, similarity_weight, fieldnorm_reader)) => {
+                if self.slop == 0 {
+                    Ok(Box::new(ExactPhraseScorer::new(
+                        term_postings_list,
+                        similarity_weight,
+                        fieldnorm_reader,
+                        self.scoring_enabled,
+                    )))
+                } else {
+                    Ok(Box::new(SlopPhraseScorer::new(
+                        term_postings_list,
+                        similarity_weight,
+                        fieldnorm_reader,
+                        self.scoring_enabled,
+                        self.slop,
+                    )))
+                }
+            }
```

Comment:
duplicates https://github.com/quickwit-oss/tantivy/pull/1414/files#diff-b46d170c28bc9075958011ee0f1c172d00e2a39169d470250f10151ba7588d3bR50-R67 which is in teh same file but we need to return a different Boxed trait